service/src/test/java/com/epam/esm/CertificateServiceImplTest.java

```diff
@@ -0,0 +1,270 @@
+package com.epam.esm;
+
+import com.epam.esm.dto.CertificateDTO;
+import com.epam.esm.dto.converter.CertificateConverterDTO;
+import com.epam.esm.dto.query.CertificatePageQueryDTO;
+import com.epam.esm.entity.Certificate;
+import com.epam.esm.exception.EntityAlreadyExistsException;
+import com.epam.esm.exception.EntityNotFoundException;
+import com.epam.esm.exception.ValidatorException;
+import com.epam.esm.repository.CertificateRepository;
+import com.epam.esm.repository.TagRepository;
+import com.epam.esm.service.impl.CertificateServiceImpl;
+import com.epam.esm.specification.CertificateAllSpecification;
+import com.epam.esm.validator.CertificateDTOValidator;
+import com.epam.esm.validator.CertificatePageQueryDTOValidator;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class CertificateServiceImplTest {
+    @Test
+    void testGetByIdShouldReturnCertificateDTOWhenEntityExists() {
+        Certificate certificate = new Certificate.Builder(
+                "test",
+                "test description",
+                BigDecimal.valueOf(10),
+                10)
+                .id((long) 1).build();
+        CertificateRepository certificateRepository = Mockito.mock(CertificateRepository.class);
+        TagRepository tagRepository = Mockito.mock(TagRepository.class);
+        CertificateDTOValidator certificateDTOValidator = Mockito.mock(CertificateDTOValidator.class);
+        CertificatePageQueryDTOValidator pageQueryDTOValidator = Mockito.mock(CertificatePageQueryDTOValidator.class);
+
+        Mockito.when(certificateRepository.getById(Mockito.anyLong())).thenReturn(Optional.of(certificate));
+        CertificateServiceImpl service = new CertificateServiceImpl(certificateRepository, tagRepository
+                , certificateDTOValidator, pageQueryDTOValidator);
+        CertificateDTO expected = CertificateConverterDTO.convertToDto(certificate);
+
+        CertificateDTO actual = service.getById((long) 1);
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void testGetByIdShouldThrowExceptionWhenEntityNotFound() {
+        CertificateRepository certificateRepository = Mockito.mock(CertificateRepository.class);
+        TagRepository tagRepository = Mockito.mock(TagRepository.class);
+        CertificateDTOValidator certificateDTOValidator = Mockito.mock(CertificateDTOValidator.class);
+        CertificatePageQueryDTOValidator pageQueryDTOValidator = Mockito.mock(CertificatePageQueryDTOValidator.class);
+
+        Mockito.when(certificateRepository.getById(Mockito.anyLong())).thenReturn(Optional.empty());
```

Comment:
You could use static import here to increase readability, same for Mockito.verify, Mockito.anyLong(), Mockito.anyString() and other similar methods